lib/resque/scheduler.rb

```diff
@@ -61,14 +69,19 @@ def run
               if am_master != was_master
                 procline am_master ? 'Master scheduler' : 'Child scheduler'
 
-                # Load schedule because changed
                 reload_schedule!
+                if am_master
+                  log!("Assuming master role.")
+                  # It's a protected method for some reason.
+                  rufus_scheduler.__send__(:start)
+                else
+                  log!("Assuming non-master role, not running cron jobs.")
+                  @rufus_scheduler&.shutdown(:wait)
+                end
               end
 
-              if am_master
-                handle_delayed_items
-                update_schedule if dynamic
-              end
+              handle_delayed_items
+              update_schedule if am_master && dynamic
```

Comment:
This chunk is key. Here's where we (a) trade off the responsibility for running cron jobs, so that only one daemon will run them at a time, and if they die another daemon will pick up the role, and (b) have all daemons execute jobs, not just the master.